carbonmark-api/src/routes/projects/projects.utils.ts

```diff
@@ -0,0 +1,139 @@
+import { FastifyInstance } from "fastify";
+import { map } from "lodash/fp";
+
+import { FindProjectsQueryVariables } from "src/.generated/types/marketplace.types";
+import { PoolPrice } from "src/utils/helpers/fetchAllPoolPrices";
+import { extract, notNil } from "../../utils/functional.utils";
+import {
+  getAllCategories,
+  getAllCountries,
+  getAllVintages,
+} from "../../utils/helpers/utils";
+import { POOL_INFO } from "./projects.constants";
+import {
+  FindQueryOffset,
+  FindQueryProject,
+  GetProjectResponse,
+} from "./projects.types";
+
+/**
+ * Build a default FindProjectsQueryVariables object for searching
+ * via findProjects
+ * @param fastify
+ * @returns
+ */
+export const getDefaultQueryArgs = async (
+  fastify: FastifyInstance
+): Promise<FindProjectsQueryVariables> => {
+  //Fetch all possible parameter values
+  const [category, country, vintage] = await Promise.all([
+    getAllCategories(fastify).then(map(extract("id"))),
+    getAllCountries(fastify).then(map(extract("id"))),
+    getAllVintages(fastify),
+  ]);
+  return {
+    category,
+    country,
+    vintage,
+    search: "",
+  };
+};
+
+export const buildOffsetKey = ({
+  projectID,
+  vintageYear,
+}: Pick<FindQueryOffset, "projectID" | "vintageYear">) =>
+  projectID + "-" + vintageYear;
+
+export const buildProjectKey = ({
+  registry,
+  projectID,
+  vintage,
+}: Pick<FindQueryProject, "registry" | "projectID" | "vintage">) =>
+  registry + "-" + projectID + "-" + vintage;
+
+export const getOffsetTokenPrices = (
+  offset: FindQueryOffset,
+  poolPrices: Record<string, PoolPrice>
+) => {
+  const prices: string[] = [];
+  if (parseFloat(offset.balanceUBO) >= 1) {
+    const isDefault =
+      POOL_INFO.ubo.defaultProjectTokenAddress.toLowerCase() ===
+      offset.tokenAddress.toLowerCase();
+    const priceKey = isDefault ? "defaultPrice" : "selectiveRedeemPrice";
+    prices.push(poolPrices.ubo[priceKey]);
+  }
+  if (parseFloat(offset.balanceNBO) >= 1) {
+    const isDefault =
+      POOL_INFO.nbo.defaultProjectTokenAddress.toLowerCase() ===
+      offset.tokenAddress.toLowerCase();
+    const priceKey = isDefault ? "defaultPrice" : "selectiveRedeemPrice";
+    prices.push(poolPrices.nbo[priceKey]);
+  }
+  if (parseFloat(offset.balanceNCT) >= 1) {
+    const isDefault =
+      POOL_INFO.nct.defaultProjectTokenAddress.toLowerCase() ===
+      offset.tokenAddress.toLowerCase();
+    const priceKey = isDefault ? "defaultPrice" : "selectiveRedeemPrice";
+    prices.push(poolPrices.nct[priceKey]);
+  }
+  if (parseFloat(offset.balanceBCT) >= 1) {
+    const isDefault =
+      POOL_INFO.bct.defaultProjectTokenAddress.toLowerCase() ===
+      offset.tokenAddress.toLowerCase();
+    const priceKey = isDefault ? "defaultPrice" : "selectiveRedeemPrice";
+    prices.push(poolPrices.bct[priceKey]);
+  }
+  return prices;
```

Comment:
Would this get rid of the `@todo` or is there more nuance to the conversion?

```suggestion
  return prices.map(price => Number(price));
```

Reply:
Yea I'm trying to avoid changing the response type/content as much as possible to know that my refactor has not broken anything